server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -1258,7 +1265,7 @@ public Map<String, NamedAnalyzer> indexAnalyzers() {
 
     @Override
     public FieldMapper.Builder getMergeBuilder() {
-        return new Builder(simpleName(), indexCreatedVersion, indexAnalyzers).init(this);
+        return new Builder(simpleName(), indexCreatedVersion, indexAnalyzers, false).init(this);
```

Comment:
Always providing `false` here is incorrect. It could disable synthetic source at some point. I think we need record in `TextFieldMapper` class, that synthetic source was enabled, so the right value can be provided here.

Reply:
I don't think it can actually disable it since the default value should be already set correctly before that and `init` will pick it up. I do agree it is more clear to store it though.